### PR TITLE
[MIA-277] Make changes to public data script to pull Urgent Waiver data

### PR DESCRIPTION
**Pull request by kennethedwards-bixal**
## Description
https://cm-jira.usa.gov/browse/MIA-277
This PR is for the implementation of bringing  urgent waiver data to the made-in-america-data.

## Additional information
The URL used in this PR for urgent waivers is in the portal-test, not the actual URL we would use for production.

Refactor Notes: 
1.  I created a new file called `CompileUrgentWaiver.js` that will be responsible for running the script for the urgent waivers. In the this and file `CompileJsonApp.js` I refactored the code to  pass in files, keys, and URLs so we can reuse the functions in the Datascript instead of having to re-write a whole new DataScript file just for urgent waivers. 

2.  I created a new file called under the `StringReplace` folder called `urgent-index.js`. This file will parse the urgent waiver forms data to a user friendly format.  I did it this way because if it ran under the `index.js` file used for nonavailability, the data would append properties to the urgent waiver that are not needed such as ombDeterimation, requestStatus, and waiverCoverage to name a few. Such properties would all return `N/A`. 

To differentiate the two types of files, I created an if statement on line 134 in DataScript/index.js. This if statement looks for the property wavierType(something that Nonavailability waivers don't have) and points those files to the `urgent-index.js` which manipulates urgent waiver strings. I think this works well because if ever need new urgent waiver specific properties, we can just add them to this file. I worked with Stephanie to double check public data fields.

3. I wrote a new set of tests. I copied the bulk of the `compileJsonApp.spec.js` format. I made sure that the proper urgent waivers _ids were present but more importantly I made sure that waivers were NOT removed during the script as the remove function is part of the overall processData function. Additionally, string manipulation was also tested. 

Lastly, I chose not to edit the yml file that runs the scripts till we get the proper URL.

## How to test
 - comment out line 18 in `CompileJSonApp.js` and comment out line 22 in `CompileUrgentWaiver.js`
- `npm run test`

### CODE REVIEW

File: src/Utility/StringReplace/urgent-index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
```
Comment (jefffortune) [inline src/Utility/StringReplace/urgent-index.js:1]: You can add uncertain, Uncertain to the  StringReplace/index.js switch statement to reduce overall code.

File: test/CompileUrgentWaiver.js.spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// eslint-disable-next-line node/no-unpublished-require
```
Comment (jefffortune) [inline test/CompileUrgentWaiver.js.spec.js:1]: Can combine these to CompileJSONAppTest, it testing the same things. Remove this file will help with DRY

File: CompileUrgentWaiver.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+if (process.env.NODE_ENV !== 'production') {
```
Comment (jefffortune) [inline CompileUrgentWaiver.js:1]: Combine this with the cCompieJsonApp.js file. This way circle CI doesnt have to change.

**Review by jefffortune — CHANGES_REQUESTED** [review]

**Comment by kennethedwards-bixal** [comment]
Made changes, ran tests programmatically and visually checked as well.

**Pull request closed without merging**